backend/src/routes/post.ts

```diff
@@ -35,33 +39,169 @@ router.post('/', async (ctx, next) => {
 	post.bodyText = bodyText;
 	post.musicURL = musicURL;
 
-	await post.save();
-	ctx.body = post;
+	await Post.create(post);
+	ctx.body = await post.save();
 });
 router.post('/:postID/like', async (ctx, next) => {
 	const post = await Post.findOneBy(mongodb.ObjectId(ctx.params.postID));
+
+	const requestBody = ctx.request.body as any;
+	const username = requestBody.username;
+	const dir = parseInt(requestBody.dir);
+
 	if (post === null) {
+		console.log(`Could not find post with ID: ${ctx.params.postID}`);
 		ctx.status = 404;
 		return;
 	}
-	post.numLikes += 1;
-	await post.save();
+	// currently, we have no way of identifying based off the request given
```

Comment:
now that auth is merged, you can update these routes to grab the user from the auth header instead